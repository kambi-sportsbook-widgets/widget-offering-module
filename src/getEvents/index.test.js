import fs from 'fs'
import path from 'path'
import { getEvents } from './index'
import {
  getMultiEventIdFromKambi,
  expectedEventProps,
  expectedBetOfferProps,
} from '../testHelpers'

jest.setTimeout(30000) // need to increase default timeout for functions as we're making network requests

describe('Test suite for getEvent(eventId) offering module function', () => {
  let response = null
  let eventIds = null

  beforeAll(() => {
    response = getMultiEventIdFromKambi
  })

  it(`it gets event ${eventIds} from kambi offering API`, () => {
    return response().then(ids => {
      eventIds = ids

      return getEvents(eventIds).then(data => {
        expect(data).toBeDefined()
        data.map(ev => {
          expectedEventProps.forEach(prop => expect(ev).toHaveProperty(prop))
          ev.betOffers.forEach(bo => {
            expect(bo.eventId).toEqual(ev.id)
            expectedBetOfferProps.forEach(prop =>
              expect(bo).toHaveProperty(prop)
            )
          })
        })
      })
    })
  })

  it('it tries to get an event with an invalid event ID', () => {
    return getEvents(['123123', 'test']).catch(err => {
      expect(err).toMatchSnapshot()
    })
  })

  it('it tries to get an event without passing an event id', () => {
    expect(() => getEvents()).toThrowErrorMatchingSnapshot()
  })
})
