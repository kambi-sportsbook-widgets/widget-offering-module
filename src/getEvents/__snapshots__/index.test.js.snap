// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for getEvent(eventId) offering module function it tries to get an event with an invalid event ID 1`] = `
Object {
  "body": "{\\"error\\":{\\"status\\":406,\\"message\\":\\"Not acceptable format\\"}}",
  "status": 406,
  "statusText": "Not Acceptable",
}
`;

exports[`Test suite for getEvent(eventId) offering module function it tries to get an event without passing an event id 1`] = `"Cannot read property 'length' of undefined"`;
